colossalai/tensor/colo_tensor.py

```diff
@@ -31,21 +40,29 @@ def __new__(cls, data: torch.Tensor, spec: TensorSpec = TensorSpec(distspec.repl
         return torch.Tensor._make_subclass(cls, data, data.requires_grad)
 
     def __init__(self, data: torch.Tensor, spec: TensorSpec = TensorSpec(distspec.replicate())) -> None:
-        self._spec = copy(spec)
+        """__init__ of the tensor. It worth noting that the class's base class torch.Tensor 
+        dose not have a __init__ function. We first set the spec.
+        Args:
+            data (torch.Tensor): a torch tensor used as the payload of the tensor.
+            spec (TensorSpec, optional): the tensor spec of initialization. Defaults to TensorSpec(distspec.replicate()).
+        """
+        self._tensor_spec = copy(spec)
         self._type = TensorType.NONMODEL
+        self._tensor_spec = TensorSpec(distspec.replicate())
```

Comment:
TensorSpec init twice.

Reply:
OK, I fixed it. Since I suppose the tensor arg to init a shard tensor is the global tensor.